inventory/communication.js

```diff
@@ -0,0 +1,220 @@
+        if (filename === "inventory-data.zip") {
+            const writeFilePromise = promisify(writeFile);
+            const buffer = Buffer.from(data, "base64");
+            const zipPath = "public/inventory-data.zip";
+            await writeFilePromise(zipPath, buffer);
+            await anzip(zipPath, { outputPath: "public/inventory-data", outputContent: true });
+            console.log("Synced resources");
+        } else {
+            console.log("Got resource which isn't inventory-data.zip. Skipping to sync...");
+        }
+    }
+
+    addConnection(sock) {
+        this.connections.push(sock);
+
+        sock.on("message", (msg) => this.handleMessage(msg));
+        sock.on("close", () => this.removeConnection(sock));
+        this._sendInitialState(sock);
+    }
+
+    removeConnection(sock) {
+        const index = this.connections.indexOf(sock);
+        if (index >= 0) {
+            this.connections.splice(index, 1);
+        }
+    }
+
+    handleMessage(messageData) {
+        try {
+            const message = JSON.parse(messageData);
+            console.log(`Handling command ${message.command}`);
+            switch (message.command) {
+                case "req_component_info":
+                    this._handleComponentInfoRequest();
+                    break;
+                case "report_checksums":
+                    this._handleReportChecksumsRequest();
+                    break;
+                case "file":
+                    this._syncFile(message.path, message.data);
+                    break;
+                case "buy":
+                    this.inventory.buy(message.item);
+                    break;
+                case "sell":
+                    this.inventory.sell(message.item);
+                    break;
+                case "add":
+                    this.inventory.add(message.item);
+                    break;
+                case "remove":
+                    this.inventory.remove(message.item);
+                    break;
+                case "change_money":
+                    this.inventory.changeMoney(message.amount);
+                    break;
+                case "enable_achievement":
+                    this.inventory.enableAchievement(message.achievement);
+                    break;
+                case "setup":
+                case "reset":
+                    this.inventory.reset();
+                    this.reloadInventory();
+                    break;
+                case "restart":
+                    this.restart();
+                    break;
+                default:
+                    console.log(`Unhandled command ${message.command}`);
+                    break;
+            }
+        } catch (e) {
+            console.log(`Got error when handling cmd: ${e}`);
+        }
+    }
+
+    reloadInventory() {
+        this.inventory.loadStaticDataFrom("public/inventory-data/inventory-data.json");
+    }
+
+    async restart() {
```

Comment:
This (of course) only works when we're using nodemon, which is a problem if we make some kind of build rather that running with make when we're live.

Or do you reckon make is fine, since this component won't need to run on the Pi?

Reply:
It would be nice with a better solution, but for now I think this is fine (but we should definitely make a ticket).

Reply:
Made an issue out of this.